### ISSUE TITLE
Appium Crash after Performing cleanup

### ISSUE THREAD

**Issue by fatahillahardhi** [appium/appium#20093]
### Do I have the most recent component updates?

- [x] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Appium always crashes after `[BaseDriver] Performing cleanup of 1 cached application` 
and shows these errors:

`Original error: Unable to launch WebDriverAgent. Original error: TypeError: Cannot perform 'get' on a proxy that has been revoked. Make sure the application com.bibit.WebDriverAgentRunner.xctrunner exists and is launchable. Refer to https://appium.github.io/appium-xcuitest-driver/latest/preparation/real-device-config/ for assistance in completing the preparation.`

Initially, I used PreinstalledWDA on iOS 17 and iOS 16, encountering these errors. Later, I attempted to use only PreinstalledWDA on iOS 16 (as I had been doing for a year without issues), but encountered the same problem.

### Expected Behavior

Appium runs smoothly despite encountering errors without crashing.

### Minimal Reproducible Example

I ran the command` appium --port 4724 --session-override --use-plugins=device-farm,appium-dashboard,gestures --plugin-device-farm-platform=ios --plugin-device-farm-ios-device-type=both` 
with 6 real devices: 3 running iOS 16 and 3 running iOS 17. It was functioning correctly before I upgraded to xcuitest 7.x.x. However, when I upgraded to v7.12.0 to implement PreInstalled, I encountered these errors (I also tried upgrading to the latest version but still encountered the same errors).

### Environment

- Operating system: mac os 14.1.1
- Appium server version (output of `appium --version`): 2.4.1
- Appium driver(s) and their version(s): xcuitest 7.12.0
- Appium plugin(s) and their version(s): appium gestures 4.0.1, appium device farm 8.2.6, appium dashboard 2.0.2
- Node.js version (output of `node --version`): 16.15.1
- `npm` version (output of `npm --version`): 8.11.0
- Last component(s) version which did _not_ exhibit the problem: 
- Platform and version under test: iOS 17 and iOS 16
- Real device or emulator/simulator: real device

### Link to Appium Logs

https://gist.github.com/fatahillahardhi/208528668f60acfde78e9ae4d491ca00

**Comment by mykola-mokhnach** [comment]
The above exception is expected for older Xcuitest driver versions. Please provide the log from the recent one.

**Comment by KazuCocoa** [comment]
Hm, I haven't checked well yet but possibly some error handling with `usePreinstalledWDA`...? The caps gave `usePreinstalledWDA` but no the app on the device.

```
[XCUITestDriver@848b (903fdf45)] Starting WebDriverAgent initialization with the synchronization key 'XCUITestDriver'
[XCUITestDriver@848b (903fdf45)] Trying to start WebDriverAgent once since at least one of webDriverAgentUrl,usePreinstalledWDA capabilities is provided
[XCUITestDriver@848b (903fdf45)] Event 'wdaStartAttempted' logged at 1715599585109 (18:26:25 GMT+0700 (Western Indonesia Time))
[XCUITestDriver@848b (903fdf45)] The bundle id 'com.facebook.WebDriverAgentRunner.xctrunner' did not exist
[XCUITest] Getting bundle ID from app '/Users/devicefarm/Documents/WDA/Bibit/WDA16/WebDriverAgentRunner-Runner.app': 'com.bibit.WebDriverAgentRunner.xctrunner'
[XCUITestDriver@848b (903fdf45)] Installing prebuilt WDA at '/Users/devicefarm/Documents/WDA/Bibit/WDA16/WebDriverAgentRunner-Runner.app'. Bundle identifier: com.bibit.WebDriverAgentRunner.xctrunner.
[XCUITestDriver@848b (903fdf45)] Installing '/Users/devicefarm/Documents/WDA/Bibit/WDA16/WebDriverAgentRunner-Runner.app' on the device with UUID '00008110-00165C180E9A201E'...
[XCUITestDriver@848b (903fdf45)] Using 'serial' app deployment strategy. You could change it by providing another value to the 'appInstallStrategy' capability
[XCUITestDriver@c7b5 (bb241dbd)] A new application installation is going to be performed. Will timeout in 466640.697167 ms
[XCUITest] Getting bundle ID from app '/Users/devicefarm/Documents/WDA/Bibit/WDA16/WebDriverAgentRunner-Runner.app': 'com.bibit.WebDriverAgentRunner.xctrunner'
[XCUITest] Successfully scanned the tree structure of '/Users/devicefarm/Documents/WDA/Bibit/WDA16/WebDriverAgentRunner-Runner.app'
[XCUITest] Got 27 folders and 39 files to push
```

I wonder if the caps needs to have `appium:updatedWDABundleId` with `com.bibit.WebDriverAgentRunner`
https://appium.github.io/appium-xcuitest-driver/latest/guides/run-preinstalled-wda/

**Comment by fatahillahardhi** [comment]
Hmm, I tried using my old WebDriverAgent (WDA) for iOS 16 (which I generated last year) and it worked well without terminating Appium, but I encountered other issues.

As I mentioned earlier, I was running with 6 devices, and each device encountered different errors (though other devices worked well without errors).

The errors were:

1. `org.openqa.selenium.NoSuchSessionException: A session is either terminated or not started`
2. `org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: An unknown server-side error occurred while processing the command. Original error: Unable to launch WebDriverAgent. Original error: Failed to start the preinstalled WebDriverAgent in 90000 ms. The WebDriverAgent might not be properly built or the device might be locked. The 'appium:wdaLaunchTimeout' capability modifies the timeout. Make sure the application com.bibit.WebDriverAgentRunner.xctrunner exists and is launchable. Refer to https://appium.github.io/appium-xcuitest-driver/latest/preparation/real-device-config/ for assistance with completing the preparation.`
3. `org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Cannot read properties of null (reading 'executeAtom')`

I tried running with `appium:updatedWDABundleId`, but the result remained the same. Additionally, I upgraded both Appium to v2.5.4 and XCUITest to v7.15.2.

But mostly, the error encountered was number 1 (session terminated).

Here's the log:
- appium log (sorry if the log is too long and gets automatically truncated by my terminal): https://gist.github.com/fatahillahardhi/4b9f24340783a5a4903d82780ae4c498

cc @mykola-mokhnach and @KazuCocoa

**Comment by mykola-mokhnach** [comment]
Sorry, it is impossible to parse the large log. Please report each issue separately according to the provided template.

**Comment by fatahillahardhi** [comment]
@KazuCocoa  In the first attempt, Appium usually breaks. I tried using the pm2 library so that when it breaks, it will restart Appium. Here, sometimes it continues successfully, but mostly it ends up with an error like `org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are an invalid address of the remote server or browser startup failure.`

However, when I tried with just one device, Appium did break initially, but when Appium attempted to resume, it started running again and succeeded. However, when I tried with 3 devices, all of them running iOS 17, it still did not succeed and encountered errors

Perhaps these logs can help provide more clarity and insight into the issue.
https://gist.github.com/fatahillahardhi/35039c38eda85876aa5c785640d3f2eb
